rust/blockstore/src/memory/provider.rs

```diff
@@ -27,7 +27,11 @@ impl MemoryBlockfileProvider {
 
     pub(crate) fn open<
         'new,
-        K: Key + Into<KeyWrapper> + From<&'new KeyWrapper> + ArrowReadableKey<'new> + 'new,
+        K: Key
+            + Into<KeyWrapper>
+            + TryFrom<&'new KeyWrapper, Error = &'static str>
```

Comment:
I would prefer if we used our Error pattern here

Reply:
ACK